### PR TITLE
[NFTs] Allow to set the role to None

### PR DESCRIPTION
**Pull request by jsidorenko**
This PR allows setting the Issuer/Admin/Freezer accounts to `None` without the ability to change it back (except for root calls).
This is useful for cases when the collection is created, all the items are minted and now we want to be 100% sure no one will mint any more items, or change the metadata, or disable some item's transfer.

### CODE REVIEW

File: frame/nfts/src/features/roles.rs [hunk 1]
```diff
@@ -23,24 +23,46 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
+			let is_root = maybe_check_owner.is_none();
 			if let Some(check_origin) = maybe_check_owner {
 				ensure!(check_origin == details.owner, Error::<T, I>::NoPermission);
 			}
 
-			// delete previous values
-			Self::clear_roles(&collection)?;
-
-			let account_to_role = Self::group_roles_by_account(vec![
+			let roles_map = vec![
```
Comment (gilescope) [inline frame/nfts/src/features/roles.rs:37]: This could just be a fixed array?

Comment (jsidorenko) [reply frame/nfts/src/features/roles.rs]: done

File: frame/nfts/src/features/create_delete_item.rs [hunk 2]
```diff
@@ -123,27 +120,28 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
...
-		if !metadata.len().is_zero() {
-			Self::do_set_item_metadata(
-				Some(origin.clone()),
-				collection,
-				item,
-				metadata,
-				Some(mint_to.clone()),
-			)?;
+		let admin_account = Self::find_account_by_role(&collection, CollectionRole::Admin);
+		if let Some(admin_account) = admin_account {
```
Comment (gilescope) [inline frame/nfts/src/features/create_delete_item.rs:124]: just wondering if admin_account is_none && we are root that ought to be allowed but I'm not sure that root would ever be interested in doing so?

**Review by gilescope — APPROVED** [review]
Love PRs that minimise trust.

File: frame/nfts/src/features/create_delete_item.rs (hunk 2 above)
Comment (jsidorenko) [reply frame/nfts/src/features/create_delete_item.rs:124]: root won't be able to pre-sign anything, and there are other methods for the root account like force_mint()

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/2513987

**Comment by jsidorenko** [comment]
bot rebase

**Comment by ruseinov** [comment]
Curious why there isn’t any change to the logic that checks those roles, since they’re optional now - that should have reflected on the check logic?
One other question - what happens if the admin unsets itself?

**Review by ruseinov — APPROVED** [review]
looks good, would still be nice to make sure the answers to my questions make it to github for posterity.

**Comment by jsidorenko** [comment]
> Curious why there isn’t any change to the logic that checks those roles, since they’re optional now - that should have reflected on the check logic?

That's the beauty of that PR :) All the business logic is already in place and was mostly set here: https://github.com/paritytech/substrate/pull/13482, so now if you remove the Admin role and the previous admin will try to call some admin-related method, the validation won't allow that.

**Comment by jsidorenko** [comment]
> One other question - what happens if the admin unsets itself?

Only the collection's owner could unset the admin.

**Comment by jsidorenko** [comment]
bot merge

**Pull request merged**